### PR TITLE
Feature/sof 4733 - Refactoring

### PR DESCRIPTION
**Pull request by AcylSilane**

### CODE REVIEW

File: DigitalEcosystem/refined/perovskite_comparison/perovskites.py [hunk 1]
```diff
@@ -135,47 +135,39 @@ def unscale(arr):
 train_pred_y = model.predict(train_x)
 test_pred_y = model.predict(test_x)
 
-plt.scatter(x=unscale(train_pred_y), y=unscale(train_y), label="Train")
-plt.scatter(x=unscale(test_pred_y), y=unscale(test_y), label="Test")
-plt.xlabel("Predicted")
-plt.ylabel("Actual")
-plt.legend()
+def create_partiy_plot(train_yhat, test_yhat):
```
Comment (timurbazhirov) [inline DigitalEcosystem/refined/perovskite_comparison/perovskites.py]: Yay, "partiy"! Pls fix.

Any chance we can reuse the content between `save_parity_plot`, `create_partiy_plot`, `create_multi_parity_plot`?

File: DigitalEcosystem/utils/fingerprints.py [hunk 2]
```diff
@@ -73,3 +82,50 @@ def fingerprint_ewald_sum(max_atoms: int, df: pd.DataFrame,) -> np.ndarray:
     result = np.real(np.vstack(df['atoms_object (unitless)'].apply(ewald_matrix.create)))
 
     return result
+
+
+# ============
+# Descriptors
+# ============
```
Comment (timurbazhirov) [inline DigitalEcosystem/utils/fingerprints.py]: We need to explain the naming above - what is Fingerprint? Descriptor?

File: DigitalEcosystem/refined/metal_classifier/metalClassifier.py [hunk 3]
```diff
@@ -157,7 +156,7 @@ def keep_best_bg(study, trial):
 def objective(trial: optuna.Trial):
     global current_pipeline
 
-    tr_x, val_x, tr_y, val_y = sklearn.model_selection.train_test_split(res_x, res_y, test_size=0.2)
+    tr_x, val_x, tr_y, val_y = sklearn.model_selection.train_test_split(res_x, res_y, test_size=0.2, random_state=RANDOM_SEED)
```
Comment (timurbazhirov) [inline DigitalEcosystem/refined/metal_classifier/metalClassifier.py]: Let's spell the variable names, pls: tr -> train, val -> value, etc.

File: DigitalEcosystem/refined/perovskite_comparison/perovskites.py (hunk 1 above)
Comment (AcylSilane) [reply DigitalEcosystem/refined/perovskite_comparison/perovskites.py]: Sure, added

File: DigitalEcosystem/utils/fingerprints.py (hunk 2 above)
Comment (AcylSilane) [reply DigitalEcosystem/utils/fingerprints.py]: Added comments to the comments

File: DigitalEcosystem/refined/metal_classifier/metalClassifier.py (hunk 3 above)
Comment (AcylSilane) [reply DigitalEcosystem/refined/metal_classifier/metalClassifier.py]: Naming convention for these variables has been made more verbose

**Comment by AcylSilane** [comment]
Merging per https://exabyte.atlassian.net/browse/SOF-4733?focusedCommentId=42114

**Pull request merged**